tests/integration_tests/synthetic_test_data.py

```diff
@@ -0,0 +1,71 @@
+"""Utilities for producing synthetic test data that is convergence-friendly."""
+
+from collections import namedtuple
+
+import numpy as np
+import pandas as pd
+from sklearn.model_selection import train_test_split
+
+RANDOM_SEED = 42
+NUMBER_OBSERVATIONS = 500
+
+GeneratedData = namedtuple("GeneratedData", "train_df validation_df test_df")
+
+
+def get_feature_configs():
+    input_features = [
+        {"name": "x", "type": "number"},
+    ]
+    output_features = [
+        {
+            "name": "y",
+            "type": "number",
+            "loss": {"type": "mean_squared_error"},
+            "decoder": {
+                "num_fc_layers": 5,
```

Comment:
nit: Let's make this smaller so the tests that use synthetic data run quicker, I'd probably use 1 or 2 at most

Reply:
Done.